core/middlewares/with-analytics-cookies.ts

```diff
@@ -0,0 +1,24 @@
+import { NextRequest } from 'next/server';
+
+import { setVisitIdCookie, setVisitorIdCookie } from '~/lib/analytics/bigcommerce';
+
+import { MiddlewareFactory } from './compose-middlewares';
+
+export const withAnalyticsCookies: MiddlewareFactory = (next) => {
+  return async (request, event) => {
+    const visitorId = await setVisitorIdCookie();
+    const { id: visitId, isNew: isVisitNew } = await setVisitIdCookie();
+
+    if (isVisitNew) {
+      event.waitUntil(sendShopperVisitEvent(request, visitorId, visitId));
+    }
+
+    return next(request, event);
```

Comment:
The more I think about this, I think the `set*` functions shouldn't contain side effects 🤔 Can we move the logic into the middleware instead of magically handling it inside the fns?
```suggestion
    let visitorId = await getVisitorIdCookie();
    let visitId = await getVisitIdCookie();
    
    if (!visitorId) {
      visitorId = await setVisitorIdCookie();
    }

    if (!visitId) {
      visitId = await setVisitIdCookie();

      event.waitUntil(sendShopperVisitEvent(request, visitorId, visitId));
    }

    return next(request, event);
```